cmd/podman/create.go

```diff
@@ -245,10 +269,13 @@ func parseCreateOpts(ctx context.Context, c *cli.Context, runtime *libpod.Runtim
 		return nil, err
 	}
 
-	imageID := data.ID
+	imageID := ""
 
-	if len(c.Args()) > 1 {
+	if data == nil {
+		inputCommand = c.Args()
+	} else {
 		inputCommand = c.Args()[1:]
```

Comment:
This works, but the interaction between `data` and `c.Args()` will be very non-obvious after without the context of this PR. Could the `inputCommand` determination be moved to the caller’s `if rootfs == ""` condition, or maybe the `if rootfs == ""` condition into this function?

Reply:
I am still unsure about how the CLI should look like, the image argument doesn't make any sense when rootfs is specified (as the current PR does) but I am not yet sure this is the nicest way to handle it.  @rhatdan @mheon @baude @wking what do you think?

Reply:
What if the last field was an image or a path?  The path would then be the rootfs.

Use path for rootfs and skip all image handling?
podman run -ti ~/mycontainer /bin/sh

versus

podman run -ti fedora /bin/sh

But why can't we get the second example to work with containers/storage, Just have podman realize it is not running with CAP_SYS_ADMIN and so it sets up a local storage ~/containers/storage and pulls the images to a vfs layer (Hopefully at some point an overlayfs, if UserNS gets support).

Reply:
so do we want `--rootfs` to not accept an argument but change the behaviour of the first argument?

podman run -ti --rootfs -v /foo:/bar --net host --other-args [...] /path/to/rootfs echo foo

vs what it is done now:

podman run -ti -v /foo:/bar --net host --other-args [...] --rootfs /path/to/rootfs echo foo

Thinking of it now, the first makes more sense to me.  As `--rootfs` can simply be "treat the image argument as a rootfs`

Reply:
unprivileged overlayfs is still WIP.  Differently FUSE is going to be soon supported in an userNS.

I've a very hacky implementation of overlay in FUSE, but I got overlayfs to work with buildah in an userNS.  I'll do some cleanup and publish it somewhere later this week.  containers/storage will need to learn how to mount an overlayfs using FUSE instead of "mount -t overlay"

Reply:
I was thinking more that you would not need --rootfs at all.  If the "image" field is a path then it implies --rootfs.

Reply:
Personally I’d prefer explicit options for deciding the semantics, instead of arguments that can be parsed multiple ways and a heuristic guess what the user meant.

Reply:
Agree with @mtrmac here. Parsing image names is an absolute mess because of the ambiguity of shortname vs fullname amongst other things. Let's not make the same mistake here.